include/geom/cell_hex.h

```diff
@@ -180,6 +180,9 @@ inline
 Hex::Hex(const unsigned int nn, Elem* p, Node** nodelinkdata) :
   Cell(nn, Hex::n_sides(), p, _elemlinks_data, nodelinkdata)
 {
+  // Make sure the interior parent isn't undefined
+  if (LIBMESH_DIM > 3)
```

Comment:
Oh, and to elaborate, "no reason to preclude it" includes "anything past -O0 will strip out the contents of an if(false) block", since we don't have any weird "the compiler has to instantiate the function even if it knows it won't reach the call" situations like we did in Antioch.